contrib/intel/jenkins/build.py

```diff
@@ -68,7 +68,7 @@ def skip(install_path):
     if os.getenv('CHANGE_TARGET') is not None:
         change_target = os.environ['CHANGE_TARGET']
     else:
-        change_target = os.environ['GIT_BRANCH']
+        change_target = 'main'
```

Comment:
What will happen if the test is run for a branch, e.g., `v1.15.x`? Will `CHANGE_TARGET` be set to `v1.15.x`?

Reply:
@j-xiong - yes.  CHANGE_TARGET should be v1.15.x.  I think the else statement above shouldn't really be hit. @zachdworkin - we're you seeing that this path was taken, or is this just a precautionary fix?

Reply:
@shefty @j-xiong The GIT_BRANCH variable is matching to whatever you named your forked branch.
For me this can be: MPI, jenkins_ci, main, etc.
MPI and jenkins_ci do not exist as tags with ofiwg/libfabric.
Changing the default change_target to 'main' will allow private jenkins jobs will allow them to actually run instead of opt-out every time for having an empty commit_id file.
This is only the case for private jobs and does not effect the PR case.

Reply:
If I understand correctly, for a pull request triggered run, CHANGE_TARGET is the branch the PR tries to merge into and GIT_BRANCH is the branch in the repo where the changes are made. Since CHANGE_TARGET is always set for PR so such run is not affected by this commit. 

For runs triggered by other means (e.g. direct push), CHANGE_TARGET is not set, bit GIT_BRANCH is. Since skip.sh always tries to compare the current tree with a branch at upstream, it has to compare with a branch that exists in order to generate a non-empty change set. Using "main" doesn't necessarily generate the actual difference (the user branch could be based on a different upstream branch) but it very likely can generate a diff that avoids the test being skipped.

More detailed description in the commit message would be helpful (you can use what I wrote above if that matches your intention). Also try to limit the length of the title line of the commit message (avoid folding over when viewed from the web, the threshold is 72 characters I believe).

Reply:
@j-xiong That is very close.

CHANGE_TARGET is always set regardless of if the run is triggered by direct push, build now, or replay but only for pull requests. 

If I were to run my private jenkins job (a non pull request job) then CHANGE_TARGET is always set to null. Even if I were to run using build now, direct push, or replay CHANGE_TARET will always be null.

You are right that using 'main' doesn't necessarily generate the actual difference between the user branch and the upstream branch however the usecase of falling into the else clause will only effect people with private jenkins jobs.
I can create a different else clause that instead of generating a (usually correct) list of changed files in this case, to instead create a list of changed files that will always decide to not opt out. That way private jobs will never opt out which is probably what we would want anyways